### PR TITLE
influxdb: add livecheck blocks for resources

### PR DESCRIPTION
**Pull request by mohammadzainabbas**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

As stated in https://github.com/Homebrew/homebrew-core/pull/111257, after https://github.com/Homebrew/brew/pull/13613, it would be possible to use `brew livecheck` command for resources as well. 

While augmenting `brew livecheck` command, I tested the functionality on multiple different resources and formulae. `pkg-config-wrapper` was one of the resource which could benefit from such behaviour.

#

> Note: you can only test this functionality properly once https://github.com/Homebrew/brew/pull/13613 is merged successfully. Simply, run `brew livecheck influxdb -r` and you'd see the livecheck output for resources as well.

# 

As pointed out by @SMillerDev and @nandahkrishna, my earlier implementation was wrong as it was looking at the latest version. However, now it will fetch `pkg-config-wrapper` livecheck info from upstream repo version in `go.mod` of `influxdb` (as intended).

### CODE REVIEW

**Comment by SMillerDev** [comment]
This still has the same issue. It doesn't check the specific influxdb version requirements.

**Comment by samford** [comment]
It's technically possible to check the `go.mod` file for a specific tag (e.g., `https://raw.githubusercontent.com/influxdata/influxdb/v2.5.1/go.mod`) but it raises the question of what version should be used in the check.

For example, if we check the `go.mod` file for the current `stable` tag in the formula (i.e., `stable.specs[:tag]`), then someone manually updating the formula (while referencing `brew livecheck -r influxdb` output) could use an old version. It would also make bumping resource versions more complicated in tools like `bump-formula-pr` because we would have to run livecheck to identify a new package version, update the formula's tag/revision, and re-run livecheck to get the resource version information with respect to the new tag. That approach would be confusing and potentially error-prone but I wanted to mention it for the sake of completeness.

It's arguably more reasonable to check the `go.mod` file for the newest version reported from livecheck (i.e., 2.6.0) to determine the newest `pkg-config-wrapper` version. That would work in this particular scenario but this is the sort of complexity that I was referring to when I said that identifying appropriate resource versions is a challenge (and that adding basic resource support to livecheck was just the tip of the iceberg).

As we've discussed previously, the issue is that a `livecheck` block's URL is defined when the formula is parsed but the version from livecheck is only available after running the main formula check (i.e., only available at runtime). We discussed this situation [on Slack] a couple of months ago and @nandahkrishna was working on a solution. He initially proposed using a standardized placeholder in URLs like this (e.g., `<FORMULA_LATEST_VERSION>`). I was in favor of the idea (it seems like the only solution with our technical constraints) and I suggested adding a `latest_version` instance variable/constant to the `livecheck` block DSL that's set to the unique placeholder string. That is to say, using `#{latest_version}` in these URLs may feel more natural to contributors/maintainers and it's easier to remember than the previously mentioned string

When we left it, @nandahkrishna was going to work on implementing this feature. As it stands, we can't create an appropriate `livecheck` block for `pkg-config-wrapper` until the aforementioned work is finished (and any changes to the `livecheck` block DSL would need to be included in a new `brew` release before we can modify related formulae).

**Comment by nandahkrishna** [comment]
Apologies for the delay here. I had some progress locally to implement the feature that would allow access to the latest version from the formula's `livecheck` in the resource `livecheck` blocks. Unfortunately I had a very busy semester but I should be able to create a PR in Homebrew/brew this weekend. Thanks for the detailed explanation @samford.

**Comment by nandahkrishna** [comment]
I've updated this PR with `livecheck` blocks for both resources. This should be ready to merge (perhaps after @samford or someone has reviewed it) since the Homebrew/brew PR has been included in the latest release.

**Review by SMillerDev — APPROVED** [review]

**Pull request closed without merging**
